doc/api/crypto.md

```diff
@@ -1786,6 +1794,14 @@ The return value `{ publicKey, privateKey }` represents the generated key pair.
 When PEM encoding was selected, the respective key will be a string, otherwise
 it will be a buffer containing the data encoded as DER.
 
+Private keys can be encrypted if the `type` is PKCS#8 or the `format` is PEM.
+If a `cipher` is specified and PKCS#8 was selected, an `EncryptedPrivateKeyInfo`
+structure will be produced. If PKCS#1 or SEC1 was selected and the `format` is
+PEM, RFC1421-style PEM-level encryption will be used. For maximum compatibility,
```

Comment:
I had to look a couple times to make sure I understand this. I suggest something like

PKCS#1, SEC1, and PKCS#8 type keys can be encrypted by using a combination of the cipher and format options. The PKCS#8 type can be used with any format to encrypt any key algorithm (RSA, EC, or DH) by specifying a cipher. PKCS#1 and SEC1 can only be encrypted by specifying a cipher when the PEM format is used.

Btw - I assume that if cipher or passphrase is provided, but the other options are such that encryption is not possible, an error will be thrown? It would be unfortunate to think you were encrypting your private keys but actually were not!